crates/iceberg/src/arrow/reader.rs

```diff
@@ -1103,6 +1104,7 @@ impl BoundPredicateVisitor for PredicateConverter<'_> {
 
             Ok(Box::new(move |batch| {
                 let left = project_column(&batch, idx)?;
+                let literal = cast_literal_if_required(Arc::clone(&literal), left.data_type())?;
```

Comment:
We are calling `Arc::clone()` on the `&literal` every time we invoke `cast_literal_if_required`. Can we move this into `cast_literal_if_required` to avoid the repetition?